### PR TITLE
Respect user profile config returned by the server

### PR DESCRIPTION
**Pull request by mspensieri**
1. Use `config.profile.uploadInterval` instead of hardcoded value for
   user props throttling
2. Do not upload user props if `config.profile.enabled` is false
3. Wait to upload user props if the user has not been created yet
4. Fix `isDirty` check being dirty too often
5. Upload any pending user props after a user is created
6. Fix `update` tests to mock `setTimeout` instead of relying on
   broken sinon timers

### CODE REVIEW

File: src/frame/js/actions/user.js [hunk 1]
```diff
@@ -34,15 +33,14 @@ export function immediateUpdate(props) {
         }
 
         lastUpdateAttempt = Date.now();
-
-        props = Object.assign({}, pendingUserProps, props);
+        console.log('Updating attempt');
```
Comment (wmora) [inline src/frame/js/actions/user.js]: 🚫

Comment (mspensieri) [inline src/frame/js/actions/user.js]: 😱

File: src/frame/js/actions/user.js [hunk 2]
```diff
@@ -96,6 +99,14 @@ export function update(props) {
     };
 }
 
+export function _resetState() {
```
Comment (lemieux) [inline src/frame/js/actions/user.js]: does it really need to be exported? if it needs to be called from the test, you can use the Rewire API to get it.

File: test/specs/actions/user.spec.js [hunk 3]
```diff
@@ -47,34 +59,77 @@ describe('User Actions', () => {
...
+                };
+
+                return mockedStore.dispatch(userActions.immediateUpdate(props)).then(() => {
+                    httpStub.should.not.have.been.called;
+                });
+            });
         });
     });
 
     // skip these untils fake timers weirdness is resolved.
```
Comment (lemieux) [inline test/specs/actions/user.spec.js]: comment should go away now

**Pull request merged**